### ISSUE TITLE
pander

### ISSUE THREAD

**Issue by farnushfarhadi** [STAT545-UBC/Discussion#141]
hi every one, 

I used pander like this: pander (test , style = 'rmarkdown'  )
It works very well in the Rstudio consule but it is not like a table in .html file. 
Does any body know how can I fix this? 

Thanks

**Comment by eacton** [comment]
Can you add your code here?  I just tried this and it worked for me.  Did you remember to load the pander library?

**Comment by farnushfarhadi** [comment]
Could you please see my repo? the results are there, Yes I loaded the libraries.

**Comment by eacton** [comment]
You need to be more specific, please.  There are multiple files in your repo.

**Comment by farnushfarhadi** [comment]
This is the function I have used for tables: It gives a continent name (like "Asia")  as input and draw the table as in Task3 of my repo. 
lifeExp_time1 <- function (continentName)
{
#print(paste0("Table of life expectancy vs time for ") , continentName)
test <- gtbl %>% filter(continent == continentName) %>% group_by(year) %>% 
summarize(trimmed_mean = mean(lifeExp , trim= 0.2), 
weighted_mean = weighted.mean(lifeExp , pop))
set.caption(paste0("life Expectancy - ", continentName)) 
pander (test , style = 'rmarkdown'  )
#kable (test , format = "markdown")
#print (continentName)
}

When I use lapply over continents by this:
lapply ( levels(gtbl$continent), lifeExp_time1 )

I expected to have multiple tables for different continents. but the results in .rmd file are as below. (they looks good in Rstudio consule! and then change the shape in .rmd file as below):
## [[1]]
## [1] "\n\n|  year  |  trimmed_mean  |  weighted_mean  |\n|:------:|:--------------:|:---------------:|\n|  1952  |     39.05      |      38.8       |\n|  1957  |     41.05      |      40.94      |\n|  1962  |     43.06      |      43.1       |\n|  1967  |     45.07      |      45.18      |\n|  1972  |     47.19      |      47.21      |\n|  1977  |     49.28      |      49.21      |\n|  1982  |     51.17      |      51.02      |\n|  1987  |     52.77      |      52.82      |\n|  1992  |      53.6      |      53.37      |\n|  1997  |     52.78      |      53.28      |\n|  2002  |     51.73      |      53.3       |\n|  2007  |     53.47      |      54.56      |\n\nTable: life Expectancy - Africa\n\n"
## attr(,"class")
## [1] "knit_asis"
## attr(,"knit_cacheable")
## [1] TRUE
## 
## [[2]]
## [1] "\n\n|  year  |  trimmed_mean  |  weighted_mean  |\n|:------:|:--------------:|:---------------:|\n|  1952  |     53.17      |      60.24      |\n|  1957  |     56.18      |      62.02      |\n|  1962  |     58.96      |      63.44      |\n|  1967  |     61.06      |      64.51      |\n|  1972  |     63.08      |      65.7       |\n|  1977  |     65.02      |      67.61      |\n|  1982  |       67       |      69.19      |\n|  1987  |     68.67      |      70.36      |\n|  1992  |     70.02      |      71.72      |\n|  1997  |     71.54      |      73.19      |\n|  2002  |     72.74      |      74.25      |\n|  2007  |      73.9      |      75.36      |\n\nTable: life Expectancy - Americas\n\n"
## attr(,"class")
## [1] "knit_asis"
## attr(,"knit_cacheable")
## [1] TRUE
## 
## [[3]]
## [1] "\n\n|  year  |  trimmed_mean  |  weighted_mean  |\n|:------:|:--------------:|:---------------:|\n|  1952  |     45.42      |      42.94      |\n|  1957  |     48.62      |      47.29      |\n|  1962  |     50.91      |      46.57      |\n|  1967  |     54.49      |      53.88      |\n|  1972  |     57.77      |      57.52      |\n|  1977  |     60.67      |      59.56      |\n|  1982  |     63.26      |      61.57      |\n|  1987  |     65.58      |      63.54      |\n|  1992  |     67.26      |      65.15      |\n|  1997  |     68.72      |      66.77      |\n|  2002  |     70.06      |      68.14      |\n|  2007  |     71.42      |      69.44      |\n\nTable: life Expectancy - Asia\n\n"
## attr(,"class")
## [1] "knit_asis"
## attr(,"knit_cacheable")
## [1] TRUE
## 
## [[4]]
## [1] "\n\n|  year  |  trimmed_mean  |  weighted_mean  |\n|:------:|:--------------:|:---------------:|\n|  1952  |     65.18      |      64.91      |\n|  1957  |     67.55      |      66.89      |\n|  1962  |     69.29      |      68.46      |\n|  1967  |     70.27      |      69.55      |\n|  1972  |     71.08      |      70.47      |\n|  1977  |      72.1      |      71.54      |\n|  1982  |     73.06      |      72.56      |\n|  1987  |     73.98      |      73.45      |\n|  1992  |     74.86      |      74.44      |\n|  1997  |     75.91      |      75.71      |\n|  2002  |     77.02      |      77.02      |\n|  2007  |     77.99      |      77.89      |\n\nTable: life Expectancy - Europe\n\n"
## attr(,"class")
## [1] "knit_asis"
## attr(,"knit_cacheable")
## [1] TRUE
## 
## [[5]]
## [1] "\n\n|  year  |  trimmed_mean  |  weighted_mean  |\n|:------:|:--------------:|:---------------:|\n|  1952  |     69.25      |      69.17      |\n|  1957  |      70.3      |      70.32      |\n|  1962  |     71.09      |      70.99      |\n|  1967  |     71.31      |      71.18      |\n|  1972  |     71.91      |      71.92      |\n|  1977  |     72.85      |      73.26      |\n|  1982  |     74.29      |      74.58      |\n|  1987  |     75.32      |      75.98      |\n|  1992  |     76.94      |      77.36      |\n|  1997  |     78.19      |      78.62      |\n|  2002  |     79.74      |      80.16      |\n|  2007  |     80.72      |      81.06      |\n\nTable: life Expectancy - Oceania\n\n"
## attr(,"class")
## [1] "knit_asis"
## attr(,"knit_cacheable")
## [1] TRUE

**Comment by eacton** [comment]
Alright!  So, pander is working nicely for calling your function once, ie. lifeExp_time1("Asia"), but is getting upset when being thrown a character vector.  Another function in the pander package pandoc.table() seems to handle this okay.

We are just learning loops, but this also will clean up your output a bit further in markdown. 
You'll notice the difference between this and your lapply output when you render it in markdown.

for (i in levels(gtbl$continent)) {
    lifeExp_time1(i)
}

**Comment by farnushfarhadi** [comment]
Thank you for reply. I checked pandoc.table , it is more look like table than kable() but still it is not working in .rmd file. 

R is a powerful language which has commands like lapply, sapply and apply that help us prevent from using loops as in lower level languages such as C++. I know it is harder to use such functions, but I prefer to use them. 

I checked the for loop too, but the tables were same as before.
